LeCroy/LeCrunch3 satellite (#6)
constellation-daq/Constellation: python/constellation/satellites/LeCroy/README.md

---
# SPDX-FileCopyrightText: 2025 Laurent Forthomme, DESY and the Constellation authors
# SPDX-License-Identifier: CC-BY-4.0 OR EUPL-1.2
title: "LeCroy"
description: "Satellite controlling a LeCroy oscilloscope using the LeCrunch library"

> ---

Reviewer: There are a few additional helpful keys (sorry, undocumented so far!), e.g. "category" would be great

Author: I'll add the `category: "Data Sender"`, I guess? As they are undocumented, not sure what is your naming convention :-)

Reviewer: Hm given that all the others are currently uncategorized maybe leave it away - sorry for the confusion!

(Looking at https://constellation.pages.desy.de/satellites/index.html)

---
# SPDX-FileCopyrightText: 2025 Laurent Forthomme, DESY and the Constellation authors
# SPDX-License-Identifier: CC-BY-4.0 OR EUPL-1.2
title: "LeCroy"
description: "Satellite controlling a LeCroy oscilloscope using the LeCrunch library"
---

## Description

This satellite uses the [LeCrunch3](https://github.com/nminafra/LeCrunch3) library by [Nicola Minafra](https://github.com/nminafra),
based on [LeCrunch2](https://github.com/BenLand100/LeCrunch2) and LeCrunch, to control and fetch each channel waveform from a `LeCroy Waverunner` series scope.
Using a TCP connection to the scope's `Waverunner` software, it provides a simplified method to start a single- or sequenced-waveform acquisition
and transmits it through the [CDTP](https://constellation.pages.desy.de/protocols/cdtp.html) to Constellation receivers.

## Requirements

> This satellite requires the `[LeCrunch3]` Python module, which can either be installed [from sources](https://github.com/nminafra/LeCrunch3)
> or using [the version packaged on Pypi](https://pypi.org/project/LeCrunch3/).

Reviewer 2: ```suggestion
This satellite requires the [lecroy] component, which can be installed with:
```

Let's keep it simple here for beginners, advanced users who want to modify the `LeCrunch3` code can use the information above
